ifrs17/Import/Importers.ipynb

```diff
@@ -310,50 +310,106 @@
+                "public async Task<Args> GetArgsFromMainAsync(IDataSet dataSet)",
                 "\n{",
                 "\n    var mainTab = dataSet.Tables[Main];",
                 "\n    if(mainTab == null) ApplicationMessage.Log(Error.NoMainTab);",
                 "\n    if(mainTab.Rows.Count() == 0) ApplicationMessage.Log(Error.IncompleteMainTab);",
-                "\n    if(mainTab.Columns.FirstOrDefault(x => x.ColumnName == nameof(ReportingNode)) == null) ApplicationMessage.Log(Error.ReportingNodeInMainNotFound);",
+                "\n    if(mainTab.Columns.Where(x => x.ColumnName == nameof(Args.Year)).Count()!=1) ApplicationMessage.Log(Error.YearInMainNotFound);",
+                "\n    if(mainTab.Columns.Where(x => x.ColumnName == nameof(Args.Month)).Count()!=1) ApplicationMessage.Log(Error.MonthInMainNotFound);",
                 "\n    if(ApplicationMessage.HasErrors()) return null;",
-                "\n    ",
+                "\n",
                 "\n    var main = mainTab.Rows.First();",
-                "\n    var reportingNode = (string)main[nameof(ReportingNode)];",
-                "\n    var scenario = mainTab.Columns.Where(x => x.ColumnName == nameof(Scenario)).Count()>0? (string)main[nameof(Scenario)] : default(string);",
-                "\n   ",
+                "\n    var scenario = mainTab.Columns.Where(x => x.ColumnName == nameof(Args.Scenario)).Count()>0? (string)main[nameof(Scenario)] : default(string);",
+                "\n    var year = (int)Convert.ChangeType(main[nameof(Args.Year)], typeof(int));",
+                "\n    var month = (int)Convert.ChangeType(main[nameof(Args.Month)], typeof(int));",
+                "\n    return new Args(year, month, scenario);",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task<string> GetReportingNodeFromMainAsync(IDataSet dataSet)",
+                "\n{",
+                "\n    var mainTab = dataSet.Tables[Main];",
+                "\n    if(mainTab.Columns.FirstOrDefault(x => x.ColumnName == nameof(ReportingNode)) == null) ApplicationMessage.Log(Error.ReportingNodeInMainNotFound);",
                 "\n    if(ApplicationMessage.HasErrors()) return null;",
-                "\n",
-                "\n    ImportArgs args;",
+                "\n    return (string)mainTab.Rows.First()[nameof(ReportingNode)];",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task<PartitionedArgs> GetPartitionedArgsFromMainAsync(IDataSet dataSet)",
+                "\n{",
+                "\n    var args = await GetArgsFromMainAsync(dataSet);",
+                "\n    var mainTab = dataSet.Tables[Main];",
```

Comment:
the next 3 lines are not needed. It's a repetition of GetReportingNodeFromMainAsync, no?